packages/consumption/src/modules/requests/itemProcessors/utility/validateAttributeMatchesWithQuery.ts

```diff
@@ -0,0 +1,198 @@
+                return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The tags of the provided IdentityAttribute do not contain at least one queried tag."));
+            }
+        }
+    }
+
+    return ValidationResult.success();
+}
+
+function validateAttributeMatchesWithRelationshipAttributeQuery(
+    query: RelationshipAttributeQuery,
+    attribute: IdentityAttribute | RelationshipAttribute,
+    recipient: CoreAddress
+): ValidationResult {
+    if (!(attribute instanceof RelationshipAttribute)) {
+        return ValidationResult.error(
+            CoreErrors.requests.attributeQueryMismatch("The provided Attribute is not a RelationshipAttribute, but a RelationshipAttribute was queried.")
+        );
+    }
+
+    const recipientIsAttributeOwner = recipient.equals(attribute.owner);
+    const queriedOwnerIsEmpty = query.owner.equals("");
+
+    if (!queriedOwnerIsEmpty && !query.owner.equals(attribute.owner)) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute does not belong to the queried owner."));
+    }
+
+    if (queriedOwnerIsEmpty && !recipientIsAttributeOwner) {
+        return ValidationResult.error(
+            CoreErrors.requests.attributeQueryMismatch("You are not the owner of the provided RelationshipAttribute, but an empty string was specified for the owner of the query.")
+        );
+    }
+
+    if (query.key !== attribute.key) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried key."));
+    }
+
+    if (query.attributeCreationHints.confidentiality !== attribute.confidentiality) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried confidentiality."));
+    }
+
+    if (query.attributeCreationHints.valueType !== attribute.value.constructor.name) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute is not of the queried RelationshipAttribute Value Type."));
+    }
+
+    if (!(attribute.value instanceof Consent)) {
+        if (query.attributeCreationHints.title !== attribute.value.title) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried title."));
+        }
+
+        if (query.attributeCreationHints.description !== attribute.value.description) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried description."));
+        }
+    }
+
+    return ValidationResult.success();
+}
+
+function validateAttributeMatchesWithThirdPartyRelationshipAttributeQuery(
```

Comment:
This error is thrown as `"The provided RelationshipAttribute exists in the context of a Relationship with a third party that should not be involved."` in the ReadAttributeRequestItemProcessor. It can't be thrown in this function here, as it only works with the `content` of LocalAttributes, i.e. IdentityAttributes or RelationshipAttributes. Therefore, this function can't check what's going on in the `shareInfo` of the LocalAttribute.